esrally/utils/io.py

```diff
@@ -406,6 +406,109 @@ def has_extension(file_name, extension):
     return ext == extension
 
 
+class FileOffsetTable:
+    """
+    The FileOffsetTable represents a persistent mapping from lines in a data file to their offset in bytes in the
+    data file. This helps bulk-indexing clients to advance quickly to a certain position in a large data file.
+    """
+    def __init__(self, data_file_path, offset_table_path, mode):
+        """
+        Creates a new FileOffsetTable instance. The constructor should not be called directly but instead the
+        respective factory methods should be used.
+
+        :param data_file_path: The absolute path to the data file. This file is assumed to exist at this point.
+        :param offset_table_path: The absolute path to the corresponding offset table file. Only required to exist
+                                  for read operations on the data file.
+        :param mode: The mode in which the file offset table should be opened.
```

Comment:
Can we not infer this? If an offset file is provided we can always open for read?

Reply:
never mind this will never be called directly.